src/common/tests_dmg_helpers.c

```diff
@@ -799,12 +799,12 @@ dmg_storage_query_device_health(const char *dmg_config_file, char *host,
 			json_object_to_json_string(val));
 		if (!json_object_object_get_ex(val, "storage", &storage_info)) {
 			D_ERROR("unable to extract hosts from JSON\n");
-			return -DER_INVAL;
+			D_GOTO(out, rc = -DER_INVAL);
```

Comment:
Shouldn't goto out_json in order to execute cmd_free_args(args, argcount);?

Reply:
Yes, the intention was to add out_json. Thanks for catching it

Reply:
What about the if statement in line #791-#794? I think the cid pointed out this case as #1 of 2 case:

775        if (!json_object_object_get_ex(dmg_out, "host_storage_map",
776                                       &storage_map)) {
777                D_ERROR("unable to extract host_storage_map from JSON\n");
   	CID 313074 (#1 of 2): Resource leak (RESOURCE_LEAK) [select issue]
778                return -DER_INVAL;
779        }
780
781        json_object_object_foreach(storage_map, key, val) {

Reply:
Is this new I do not see it in defect DAOS-6685?

Reply:
It's in https://scan8.coverity.com/reports.htm#v42469/p12225/fileInstanceId=39449665&defectInstanceId=11402595&mergedDefectId=313074
There are two occurrences for this issue, #1 of 2 and #2 of 2, It seems #1 of 2 I copied above maybe also need to make the same change.

Reply:
Both occurrences has been fixed.